cmd/k8s-operator/deploy/chart/values.yaml

```diff
@@ -21,6 +21,11 @@ operatorConfig:
   # Multiple tags are defined as array items and passed to the operator as a comma-separated string
   defaultTags:
     - "tag:k8s-operator"
+
+  # Set useDefaultSecrets to false if you want to use a secretProviderClass,
+  # or otherwise don't want the default secret to be used
+  useDefaultSecrets: true
+
```

Comment:
Instead of introducing this variable, can we pull our existing `volume` and `volumeMount` definitions into the values file under the new `extraVolumes`, `extraVolumeMounts`? Then folks can, if needed, override them by using the same fields. (And, if this works, then probably also would make sense to rename them `volumes`, `volumeMounts`)

Reply:
Hi!
I believe the `extraVolumes` pattern exists because of the hypothetical scenario where an update to the helmchart would depend on a change to the default volume mounts.
A user that chooses to override a `volumes` variable would need to monitor for such changes in each release to avoid unexpected issues, and since they probably won't, maintainers would need to be prepared for an influx of issues when charts with volume changes are released.

On the other hand I'm also somewhat unhappy with the ergonomics of this change. Will look for alternatives.

As for testing, I will bump that up on my schedule. So far I've only deployed the main branch of the k8s operator.

Reply:
Thanks for the explanation @paul-at-cybr that makes perfect sense.

> On the other hand I'm also somewhat unhappy with the ergonomics of this change. Will look for alternatives.

There might be a better solution, but perhaps we can just pull the oauth volume and volume mount out and put into the values file and merge with any extra volumes/extra volume mounts. Users can then remove them if they choose- this would be more work than than unsetting the `defaultSecrets` var, but feels slightly better to me as it does not introduce an extra variable